tests/unit/Rules/FactorOfTest.php

```diff
@@ -0,0 +1,133 @@
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+/**
+ * @group  rule
+ * @covers Respect\Validation\Rules\FactorOf
+ * @covers Respect\Validation\Exceptions\FactorOfException
+ */
+class FactorOfTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider providerForValidFactorOf
+     */
+    public function testValidFactorOfShouldReturnTrue($dividend, $input)
+    {
+        $min = new FactorOf($dividend);
+        $this->assertTrue($min->__invoke($input));
+        $this->assertTrue($min->check($input));
+        $this->assertTrue($min->assert($input));
+    }
+
+    /**
+     * @dataProvider providerForInvalidFactorOf
+     */
+    public function testInvalidFactorOfShouldThrowFactorOfException($dividend, $input)
+    {
+        $this->setExpectedException('Respect\Validation\Exceptions\FactorOfException', json_encode($input) . ' must be a factor of ' . json_encode($dividend));
+
+        $min = new FactorOf($dividend);
+        $this->assertFalse($min->__invoke($input));
+        $this->assertFalse($min->assert($input));
+    }
+
+    public function providerForValidFactorOf()
+    {
+        $tests = array(
+            // Run through the first few integers.
+            array(1, 1),
+            array(2, 1),
+            array(2, 2),
+            array(3, 1),
+            array(3, 3),
+            array(4, 1),
+            array(4, 2),
+            array(4, 4),
+            array(5, 1),
+            array(5, 5),
+            array(6, 1),
+            array(6, 2),
+            array(6, 3),
+            array(6, 6),
+        );
+
+        $tests = $this->generateNegativeCombinations($tests);
+
+        $tests = $this->generateStringAndFloatCombinations($tests);
+
+        return $tests;
+    }
+
+    public function providerForInvalidFactorOf()
+    {
+        $tests = array(
+            // Run through the first few integers.
+            array(3, 2),
+            array(4, 3),
+            array(5, 2),
+            array(5, 3),
+            array(5, 4),
+            // Zeros.
+            array(1, 0),
+            array(2, 0),
+        );
+
+        $tests = $this->generateNegativeCombinations($tests);
+
+        $tests = $this->generateStringAndFloatCombinations($tests);
+
+        $extra_tests = array(
+            // Non-integer values.
+            array(1, mt_rand(1, mt_getrandmax() - 1) / mt_getrandmax()),
+            array(1, uniqid('a')),
+        );
+        $tests = array_merge($tests, $extra_tests);
+
+        return $tests;
+    }
+
+    public function providerForInvalidFactorOfDividend()
+    {
+        return array(
+            // Invalid dividend, valid input.
+            array(0, 0),
+            array(0, 1),
+            array(0.5, mt_rand()),
+            array(1.5, mt_rand()),
```

Comment:
FYI, other people think that some degree of generative testing _alongside_ explicit, hand written unit tests is not a bad thing:

- https://en.wikipedia.org/wiki/QuickCheck
- https://github.com/steos/php-quickcheck
- http://www.pivotaltracker.com/community/tracker-blog/generative-testing